test/registration.js

```diff
@@ -0,0 +1,307 @@
+			_forgeAttendee.restore();
+			_saveAttendee.restore();
+			_createAttendeeProject.restore();
+			_createAttendeeExtraInfo.restore();
+			_createAttendeeEcosystemInterest.restore();
+			_createAttendeeRequestedCollaborator.restore();
+			done();
+		});
+    });
+
+    describe('findAttendeeByUser', function () {
+        var _findByUserId;
+        var testUser;
+        var nonExistentUser;
+
+        before(function (done) {
+            testUser = User.forge({ id: 1, email: 'new@example.com' });
+            nonExistentUser = User.forge({id: 2, email: 'fake@example.com'});
+            var testAttendee = Attendee.forge({id: 100, firstName: "Example", lastName: "User"});
+
+			_findByUserId = sinon.stub(Attendee,'findByUserId');
+
+			_findByUserId.withArgs(testUser.get('id')).returns(_Promise.resolve(testAttendee));
+			_findByUserId.withArgs(sinon.match.number).returns(_Promise.resolve(null));
+			done();
+		});
+        it('finds existing user',function(done){
+            var attendee = RegistrationService.findAttendeeByUser(testUser);
+            expect(attendee).to.eventually.have.deep.property("attributes.id", 100, "ID should be 100, the searched for ID")
+                .then(function(){
+                    expect(attendee).to.eventually.have.deep.property("attributes.firstName",
+                        'Example',"first name should be Example").notify(done);
+            })
+        });
+        it('throws exception after searching for non-existent user',function(done){
+            var attendee = RegistrationService.findAttendeeByUser(nonExistentUser);
+            expect(attendee).to.eventually.be.rejectedWith(errors.NotFoundError).and.notify(done);
+        });
+        after(function(done){
+            _findByUserId.restore();
+            done();
+        });
+    });
+
+    describe('findAttendeeById',function(){
+		var _findById;
+
+		before(function(done){
+            var testAttendee = Attendee.forge({id: 1, firstName: "Example", lastName: "User"});
+
+			_findById = sinon.stub(Attendee, 'findById');
+
+			_findById.withArgs(1).returns(_Promise.resolve(testAttendee));
+			_findById.withArgs(sinon.match.number).returns(_Promise.resolve(null));
+
+			done();
+		});
+		it('finds existing attendee',function(done){
+			var attendee = RegistrationService.findAttendeeById(1);
+			expect(attendee).to.eventually.have.deep.property("attributes.id", 1,"ID should be 1, the searched for ID")
+				.then(function(){
+					expect(attendee).to.eventually.have.deep.property("attributes.firstName",
+						'Example',"first name should be Example").notify(done);
+			})
+		});
+		it('throws exception after searching for non-existent attendee',function(done){
+			var attendee = RegistrationService.findAttendeeById(2);
+			expect(attendee).to.eventually.be.rejectedWith(errors.NotFoundError).and.notify(done);
+		});
+		after(function(done){
+			_findById.restore();
+			done();
+		});
+	});
+
+    describe('updateAttendee', function() {
```

Comment:
They're also (sort of) watching for different things. With the way we have related objects set up, I don't think I can differentiate between create/update for those objects (thus why we only watch for a 'save'), but you can see that in createAttendee, I watch for a 'forge' on Attendee, and in updateAttendee, I watch for a 'set' on Attendee. I agree it could be revisited.